solutions/instances/variables.tf

```diff
@@ -0,0 +1,270 @@
+  }
+}
+
+variable "cloud_monitoring_tags" {
+  type        = list(string)
+  description = "Tags associated with the IBM Cloud Monitoring instance (Optional, array of strings)."
+  default     = []
+}
+
+variable "cloud_monitoring_service_endpoints" {
+  description = "The type of the service endpoint that will be set for the IBM cloud monitoring instance."
+  type        = string
+  default     = "public-and-private" # Change it to private
+  validation {
+    condition     = contains(["public", "private", "public-and-private"], var.cloud_monitoring_service_endpoints)
+    error_message = "The specified service_endpoints is not a valid selection"
+  }
+}
+
+########################################################################################################################
+# COS variables
+########################################################################################################################
+
+variable "add_bucket_name_suffix" {
+  type        = bool
+  description = "Add random generated suffix (4 characters long) to the newly provisioned COS bucket name. Only used if not passing existing bucket. set to false if you want full control over bucket naming using the 'cos_bucket_name' & 'cos_target_bucket_name' variable."
+  default     = true
+}
+
+variable "cos_region" {
+  type        = string
+  default     = "us-south"
+  description = "The Cloud Object Storage region."
+}
+
+variable "cos_instance_name" {
+  type        = string
+  default     = "observability-cos"
+  description = "The name to use when creating the Cloud Object Storage instance."
+}
+
+variable "cos_instance_tags" {
+  type        = list(string)
+  description = "Optional list of tags to be added to Cloud Object Storage instance. Only used if not supplying an existing instance."
+  default     = []
+}
+
+variable "cos_instance_access_tags" {
+  type        = list(string)
+  description = "A list of access tags to apply to the Cloud Object Storage instance. Only used if not supplying an existing instance."
+  default     = []
+}
+
+variable "cos_bucket_name" {
+  type        = string
+  default     = "observability-cos-bucket"
+  description = "The name to use when creating the Cloud Object Storage bucket (NOTE: bucket names are globally unique). If 'add_bucket_name_suffix' is set to true, a random 4 characters will be added to this name to help ensure bucket name is globally unique."
+}
+
+variable "cos_target_bucket_name" {
+  type        = string
+  default     = "observability-cos-target-bucket"
+  description = "The name to use when creating the Cloud Object Storage bucket for cos target (NOTE: bucket names are globally unique)."
+}
+
+variable "cos_bucket_access_tags" {
+  type        = list(string)
+  default     = []
+  description = "Optional list of access tags to be added to the SCC COS bucket."
+}
+
+variable "cos_bucket_class" {
```

Comment:
which bucket? You will want separate variables for each bucket

Reply:
I have written the code in such a way that the user has to enter the minimum amount of variables for creating two buckets so currently the configuration of both the bucket is similar that is why the same class is being used here.